src/coinbase/types.ts

```diff
@@ -689,6 +713,38 @@ export type CoinbaseConfigureFromJsonOptions = {
   basePath?: string;
 };
 
+/**
+ * CoinbaseExternalAddressStakeOptions type definition.
+ */
+export type CoinbaseExternalAddressStakeOptions = {
+  /**
+   * The mode type that you're trying to stake with.
+   * e.g.
+   */
+  mode?: StakeOptionsMode;
+
+  /**
+   * The amount to stake, unstake, or claim_stake for in a staking operation.
+   */
+  amount?: string;
+};
+
+/**
+ * CoinbaseWalletAddressStakeOptions type definition.
+ */
+export type CoinbaseWalletAddressStakeOptions = {
```

Comment:
can we not use this and instead take mode as an explicit param input ?

Reply:
Followed the suggestion on having mode param and removed this.